text/0000-route-model-name-property.md

```diff
@@ -0,0 +1,149 @@
+- Start Date: 2019-02-01
+- Relevant Team(s): (fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies)
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking:
+
+# Route Model Name Property
+
+## Summary
+
+Add a property, `modelName`, to the route that will be the key that the model hook's return value is assigned to in the controller.
+
+## Motivation
+
+The default key for the routes model hook return on the controller is `model`. This name can be vague. The name is only expressive in the case where the route/controller naming represents the resource, e.g. `foo` route returns a `foo` model instance, and relies on the developer being aware of the convention, and for no other developer to ignore (circumvent?) the convention and return something else in the model hook. By adding a `modelName` property we allow developer to be expressive, and it will help future developers understand explicitly in the controller what type of model they are working with without having to refer back to the model hook.
+
+Additionally it adds some context to the current behaviour of assigning the model hook return to the controller, which is not transparent
+
+## Detailed design
+
+Currently if a developer wants to assign a custom name for the routes model value in the controller they must do this:
+
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('articles', model);
```

Comment:
```suggestion
    controller.set('posts', model);
```

I think using **both** `articles` and `posts` makes this more complicated than it needs to be?

Reply:
Ah yeah, I get your point, I forgot that you don't even get `posts` as the default.